### PR TITLE
Don't install gstly by default on Windows platforms

### PR DESCRIPTION
**Pull request by BryceBeagle**
This allows us to install using Poetry from an msys/mingw environment where we've built gstly manually

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,7 +24,10 @@ readerwriterlock = "^1.0.6"
 pycryptodomex = "^3.9.6"
 pendulum = "^2.0.5"
 dataclasses = { version = "^0.7", python = ">=3.6,<3.7" }
-gstly = "^0.1.5"
+gstly = [
+    { version = "^0.1.6", markers = "sys_platform != 'win32'" },
+    { version = "^0.1.6", markers = "sys_platform == 'win32'", optional = true },
```
Comment (apockill) [inline pyproject.toml:29]: and just to confirm, it wasn't possible to put a git dependency here for sys_platform==win32, or you decided against it?

Comment (BryceBeagle) [reply pyproject.toml:29]: As this is the open-source repo, I deemed that to not be a good solution. No one, even linux users, would be able to work with the poetry stuff

Comment (apockill) [reply pyproject.toml:29]: Wait, why wouldn't it have worked for linux users? That's odd

Comment (BryceBeagle) [reply pyproject.toml:29]: They wouldn't be able to do things like update the lock file without having an SSH key

Comment (apockill) [reply pyproject.toml:29]: Are you sure that's not the case with having it in the `--extras`? I don't see why poetry would suddenly not need to investigate those dependencies

Comment (BryceBeagle) [reply pyproject.toml:29]: The lock contains _everything_, even things in extras. If you notice, I didn't put the git dependency in extras. I'm just relying on our build system doing in manually

**Pull request merged**